src/admin.rs

```diff
@@ -106,312 +101,549 @@ impl AdminScClient {
+                Ok(PartitionMetadataWrapper(partition))
+            } else {
+                Err(FluvioError::Other("failed to find partition".to_owned()))
+            }
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
     }
 
     #[node_bindgen]
-    async fn delete_topic(&self,topic: String) -> Result<String,ClientError> {
+    async fn list_partitions(&self) -> Result<ArrayBuffer, FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            let partitions = client.list::<PartitionSpec, _>(vec![]).await?;
+            let json_slice = serde_json::to_vec(&partitions).map_err(|err| {
+                FluvioError::Other(format!("serialization error: {}", err.to_string()))
+            })?;
+            Ok(ArrayBuffer::new(json_slice))
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
+    }
 
-        let client = self.inner.as_ref().unwrap().clone();
-        let mut client_w = client.write().await;
+    #[node_bindgen]
+    async fn create_custom_spu(
+        &self,
+        name: String,
+        spec: CustomSpuSpecWrapper,
+    ) -> Result<(), FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            client.create(name, false, spec.0).await?;
+            Ok(())
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
+    }
 
-        client_w.delete_topic(&topic).await
+    #[node_bindgen]
+    async fn delete_custom_spu(&self, key: CustomSpuKeyWrapper) -> Result<(), FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            client.delete::<CustomSpuSpec, _>(key.0).await?;
+            Ok(())
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
     }
 
+    #[node_bindgen]
+    async fn create_managed_spu(
+        &self,
+        name: String,
+        spec: SpuGroupSpecWrapper,
+    ) -> Result<(), FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            client.create::<SpuGroupSpec>(name, false, spec.0).await?;
+            Ok(())
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
+    }
 
     #[node_bindgen]
-    async fn list_spu(&self) -> Result<ArrayBuffer,ClientError> {
+    async fn delete_managed_spu(&self, name: String) -> Result<(), FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            client.delete::<SpuGroupSpec, _>(name).await?;
+            Ok(())
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
```

Comment:
Uninitialized client, not failed to lock

Reply:
commit 0cc483b fixes;